### PR TITLE
[cloud-provider-azure] StorageClasses for disks > 4 TiB

### PR DESCRIPTION
**Pull request by zuzzas**
## Description

Provide StorageClasses for disks large than 4 TiB.

## Why do we need it, and what problem does it solve?

These SCs require cachingMode to be `None`.

## Changelog entries

```changes
section: cloud-provider-azure
type: feature
summary: Provide new default StorageClasses for disks large than 4 TiB.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/azure'...

**Review by z9r5 — APPROVED** [review]

File: modules/030-cloud-provider-azure/docs/CONFIGURATION.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ StorageClass Configuration Parameters:
 
 * `provision` — sets additional StorageClasses for [Azure ultra disks](https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#ultra-disk):
   * `name` — the name of the class to create;
+  * `type` — SKU type;
+  * `cachingMode` — string value that corresponds to destired caching mode;
```
Comment (konstantin-axenov) [inline modules/030-cloud-provider-azure/docs/CONFIGURATION.md:26]: Provide information about available values. Add link to the docs https://docs.microsoft.com/en-us/azure/virtual-machines/premium-storage-performance#disk-caching.

File: modules/030-cloud-provider-azure/docs/CONFIGURATION.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ StorageClass Configuration Parameters:
 
 * `provision` — sets additional StorageClasses for [Azure ultra disks](https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#ultra-disk):
   * `name` — the name of the class to create;
+  * `type` — SKU type;
```
Comment (konstantin-axenov) [inline modules/030-cloud-provider-azure/docs/CONFIGURATION.md:25]: Why not `skuName` to make it similar to original csi driver?

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/azure'...

File: modules/030-cloud-provider-azure/docs/CONFIGURATION.md (hunk 2 above)
Comment (zuzzas) [reply modules/030-cloud-provider-azure/docs/CONFIGURATION.md]: Done.

File: modules/030-cloud-provider-azure/docs/CONFIGURATION.md (hunk 1 above)
Comment (zuzzas) [reply modules/030-cloud-provider-azure/docs/CONFIGURATION.md:26]: Done.

**Review by z9r5 — APPROVED** [review]

File: modules/030-cloud-provider-azure/docs/CONFIGURATION.md (hunk 2 above)
Comment (konstantin-axenov) [reply modules/030-cloud-provider-azure/docs/CONFIGURATION.md]: Sorry, I think it was a bad decision to do these changes. We have type field in all other providers, so it's better to keep `type`. I suggest to change description to `azure disk storage account type`, add link to docs https://docs.microsoft.com/en-us/azure/aks/azure-disk-csi#storage-class-driver-dynamic-disk-parameters and provide available types with additional information for users https://docs.microsoft.com/en-us/azure/storage/common/storage-account-overview#types-of-storage-accounts.

File: modules/030-cloud-provider-azure/docs/CONFIGURATION.md [hunk 3]
```diff
@@ -22,6 +22,10 @@ StorageClass Configuration Parameters:
 
 * `provision` — sets additional StorageClasses for [Azure ultra disks](https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#ultra-disk):
   * `name` — the name of the class to create;
+  * `skuName` — SKU name;
```
Comment (konstantin-axenov) [inline modules/030-cloud-provider-azure/docs/CONFIGURATION.md:25]:
```suggestion
  * `type` - azure disk storage account type. Available values are `Standard_LRS`, `Premium_LRS`, `StandardSSD_LRS`, `UltraSSD_LRS`, `Premium_ZRS`, `StandardSSD_ZRS`. Check out [Azure docs](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-overview#types-of-storage-accounts) for more information.
```

File: modules/030-cloud-provider-azure/docs/CONFIGURATION_RU.md [hunk 4]
```diff
@@ -22,6 +22,10 @@ title: "Cloud provider — Azure: настройки"
 
 * `provision` — дополнительные StorageClass'ы для [Azure ultra disks](https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#ultra-disk):
   * `name` — имя будущего класса;
+  * `skuName` — имя SKU;
+  * `cachingMode` — строка, обозначающая желаемый режим кэширования;
+      Может быть `None`, `ReadOnly`, `ReadWrite`. Если размер диска превышает 4 TiB, необходимо использовать `cachineMode: None`.
+      Обратитесь к [документации Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/premium-storage-performance#disk-caching) за информацией.
```
Comment (konstantin-axenov) [inline modules/030-cloud-provider-azure/docs/CONFIGURATION_RU.md:28]:
```suggestion
      Обратитесь к [документации Azure](https://docs.microsoft.com/ru-ru/azure/virtual-machines/premium-storage-performance#disk-caching) за информацией.
```
Links must be on russian docs, if it exists.

**Review by konstantin-axenov — CHANGES_REQUESTED** [review]

File: modules/030-cloud-provider-azure/docs/CONFIGURATION.md (hunk 2 above)
Comment (zuzzas) [reply modules/030-cloud-provider-azure/docs/CONFIGURATION.md]: Done.

File: modules/030-cloud-provider-azure/docs/CONFIGURATION_RU.md [hunk 5]
```diff
@@ -22,6 +22,10 @@ title: "Cloud provider — Azure: настройки"
 
 * `provision` — дополнительные StorageClass'ы для [Azure ultra disks](https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#ultra-disk):
   * `name` — имя будущего класса;
+  * `type` — тип Azure диска. Доступные значения: `Standard_LRS`, `Premium_LRS`, `StandardSSD_LRS`, `UltraSSD_LRS`, `Premium_ZRS`, `StandardSSD_ZRS`. В [Azure документации](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-overview#types-of-storage-accounts) больше информации.
```
Comment (konstantin-axenov) [inline modules/030-cloud-provider-azure/docs/CONFIGURATION_RU.md:25]:
```suggestion
  * `type` — тип Azure диска. Доступные значения: `Standard_LRS`, `Premium_LRS`, `StandardSSD_LRS`, `UltraSSD_LRS`, `Premium_ZRS`, `StandardSSD_ZRS`. В [Azure документации](https://docs.microsoft.com/ru-ru/azure/storage/common/storage-account-overview#types-of-storage-accounts) больше информации.
```

File: modules/030-cloud-provider-azure/hooks/storage_classes [hunk 6]
```diff
@@ -37,6 +38,21 @@ function __main__() {
     {
       "name": "managed-premium",
       "type": "Premium_LRS"
+    },
+    {
+      "name": "managed-standard-ssd-huge",
```
Comment (konstantin-axenov) [inline modules/030-cloud-provider-azure/hooks/storage_classes:43]:
```suggestion
      "name": "managed-standard-ssd-large",
```

File: modules/030-cloud-provider-azure/hooks/storage_classes [hunk 7]
```diff
@@ -37,6 +38,21 @@ function __main__() {
...
+      "type": "StandardSSD_LRS",
+      "cachingMode": "None"
+    },
+    {
+      "name": "managed-standard-huge",
```
Comment (konstantin-axenov) [inline modules/030-cloud-provider-azure/hooks/storage_classes:48]:
```suggestion
      "name": "managed-standard-large",
```

File: modules/030-cloud-provider-azure/hooks/storage_classes [hunk 8]
```diff
@@ -37,6 +38,21 @@ function __main__() {
...
+      "type": "Standard_LRS",
+      "cachingMode": "None"
+    },
+    {
+      "name": "managed-premium-huge",
```
Comment (konstantin-axenov) [inline modules/030-cloud-provider-azure/hooks/storage_classes:53]:
```suggestion
      "name": "managed-premium-large",
```

File: modules/030-cloud-provider-azure/hooks/storage_classes_test.go [hunk 9]
```diff
@@ -71,6 +72,21 @@ cloudProviderAzure:
     "name": "managed-premium",
     "type": "Premium_LRS"
   },
+  {
+    "cachingMode": "None",
+    "name": "managed-standard-ssd-huge",
```
Comment (konstantin-axenov) [inline modules/030-cloud-provider-azure/hooks/storage_classes_test.go:77]:
```suggestion
    "name": "managed-standard-ssd-large",
```

File: modules/030-cloud-provider-azure/hooks/storage_classes_test.go [hunk 10]
```diff
@@ -71,6 +72,21 @@ cloudProviderAzure:
...
+    "type": "StandardSSD_LRS"
+  },
+  {
+    "cachingMode": "None",
+    "name": "managed-standard-huge",
```
Comment (konstantin-axenov) [inline modules/030-cloud-provider-azure/hooks/storage_classes_test.go:82]:
```suggestion
    "name": "managed-standard-large",
```

File: modules/030-cloud-provider-azure/hooks/storage_classes_test.go [hunk 11]
```diff
@@ -71,6 +72,21 @@ cloudProviderAzure:
...
+    "type": "Standard_LRS"
+  },
+  {
+    "cachingMode": "None",
+    "name": "managed-premium-huge",
```
Comment (konstantin-axenov) [inline modules/030-cloud-provider-azure/hooks/storage_classes_test.go:87]:
```suggestion
    "name": "managed-premium-large",
```

**Pull request merged**